File: evidence/generated-systems/golang/aerospace-on-golang/aerospace-on-golang/internal/model/MROFacility.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// MROFacility Declaration
//==============================================================
type MROFacility struct {
    gorm.Model
     Name                                    string
    ApprovalScope                                    string
    Address                                                            string
     Appointments           []MaintenanceAppointment `gorm:"foreignKey:AppointmentsFromMROFacilityId"`
     WorkOrders           []MaintenanceWorkOrder `gorm:"foreignKey:WorkOrdersFromMROFacilityId"`

// parent associations as their child

}

File: evidence/generated-systems/django/aerospace-on-django/aerospaceOnDjango/models/MROFacility.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model MROFacility
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class MROFacility Declaration
#======================================================================
class MROFacility (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	approvalScope = models.CharField(max_length=200, null=True)
	address = Address
	appointments = models.ManyToManyField('MaintenanceAppointment',  blank=True, related_name='+')
	workOrders = models.ManyToManyField('MaintenanceWorkOrder',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.approvalScope
		str = str + self.address
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "MROFacility";
    
	def objectType(self):
		return "MROFacility";
